File: api/schemas/movement.py
from pydantic import BaseModel
from datetime import date


class MovementBase(BaseModel):
    type: str
    date: date
    amount: int
    currency: str
    note: str = ""
    user_id: int
    account_id: int
    category_id: int = None


class MovementSchema(MovementBase):
    id: int

    class Config:
        orm_mode = True

File: client/src/models/movement.ts
export interface IMovement {
	id: number;
	type: string;
	date: string;
	amount: number;
	currency: string;
	note?: string;
	user_id: number;
	account_id: number;
	card_id?: number;
	category_id?: number;
}

export enum MovementType {
	Expense = 'Expense',
	Income = 'Income'
}

export type TMovementPayload = Omit<IMovement, 'id'>;
